regions/shapes/range.py

```diff
@@ -0,0 +1,72 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from astropy.coordinates import Angle, SkyCoord
+from regions.core.attributes import QuantityLength
+from ..core import SkyRegion, PixelRegion, PixCoord
+
+__all__ = ["RangePixelRegion", "RangeSphericalRegion"]
+
+inf = Angle(float("inf"), "deg")
+
+
+# TODO: needed?
+class RangePixelRegion(PixelRegion):
+    """
+    A range in pixel coordinates.
+
+    See `RangeSphericalRegion`.
+    """
+
+
+class RangeSphericalRegion(SkyRegion):
+    """
+    A range in longitude and latitude.
+
+    Use case: http://www.ivoa.net/documents/SIA/20151223/REC-SIA-2.0-20151223.html#toc12
+
+    Parameters
+    ----------
+    lon_min, lon_max : `~astropy.coordinates.Angle`
+        Range in longitude
+    lat_min, lat_max : `~astropy.coordinates.Angle`
+        Range in latitude
+    frame : `~astropy.coordinates.BaseCoordinateFrame`
+        Sky frame
+    meta : `~regions.RegionMeta` object, optional
+        A dictionary which stores the meta attributes of this region.
+    visual : `~regions.RegionVisual` object, optional
+        A dictionary which stores the visual meta attributes of this region.
+    """
+    # TODO: should we use these properties, or SkyCoord for min / max,
+    # or a center-based representation?
+    _params = ('lon_min', 'lon_max', 'lat_min', 'lat_max')
+    lon_min = QuantityLength('lon_min')
+    lon_max = QuantityLength('lon_max')
+    lat_min = QuantityLength('lat_min')
+    lat_max = QuantityLength('lat_max')
+
+    def __init__(self, lon_min=-inf, lon_max=inf, lat_min=-inf, lat_max=inf, frame="icrs", meta=None, visual=None):
+        self.lon_min = lon_min
+        self.lon_max = lon_max
+        self.lat_min = lat_min
+        self.lat_max = lat_max
```

Comment:
Python supports IEEE floating point -Inf and +Inf, so it should be possible, with care, to map these ranges exactly onto their IVOA SIAv2/SODA interpretation.  It may require a bit of thinking to handle the edge case of a region that actually covers the entire sky.

Reply:
OK, I'll add test cases.